File: spark/src/electro/exceptions/HttpException.java
package electro.exceptions;

import java.io.IOException;

public class HttpException extends IOException {
	private static final long serialVersionUID = 1L;
	private static String defaultMessage = "Server Error: Please try again";
	protected int code = 500;
	protected boolean loggable = false;
	protected String hiddenMessage = null;

	public HttpException() {
		super(defaultMessage);
	}

	public HttpException(String message) {
		super(message);
	}

	public HttpException(String message, String hiddenMessage) {
		super(message);
		this.hiddenMessage = hiddenMessage;
	}

	public HttpException(String message, int code) {
		super(message);
		this.code = code;
	}

	public int getStatusCode() {
		return code;
	}

	public String getHiddenMessage() {
		return hiddenMessage;
	}
}

File: asp.net_core/Exceptions/HttpException.cs
using Newtonsoft.Json;

namespace electro.Exceptions
{
    public class HttpException {
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public int Code { get; private set; }

        public string Message { get; private set; }

        public HttpException() {
            Code = 500;
			Message = "Server Error: Please try again";
        }

        public HttpException(int code, string message) {
            Code = code;
			Message = message;
		}
    }
}